app/models/publish_job.rb

```diff
@@ -1,7 +1,93 @@
 # frozen_string_literal: true
 
-# A publish job from Fedora
+# Model for metadata publish jobs
+#
+# The "state" field records the current step in the processing of this job. The
+# possible states are:
+#
+#   * :publish_not_submitted - A "publish_job" has been created, but the job has not
+#                             not been submitted to send a request to Plastron.
+#
+#   * :publish_pending - A "publish" request has been sent to Plastron, but
+#                       no response has been received.
+#
+#   * :publish_complete - All items in this import job have been successfully processed.
+#
+#   * :publish_imcomplete - Not all items in this import job have been successfully processed.
+#
+#   * :publish_in_progress - Returned for any other status
+#
+#   * :publish_error - An error has occurred during publish, such as the
+#                     STOMP client not being connected
 class PublishJob < ApplicationRecord
   belongs_to :cas_user
   serialize :solr_ids, Array
+
+  enum state: {
+    publish_not_submitted: 1,
+    publish_pending: 2,
+    publish_in_progress: 3,
+    publish_complete: 4,
+    publish_incomplete: 5,
+    publish_error: 6
```

Comment:
Plastron also returns `unpublish_complete`, `unpublish_incomplete`, and `unpublish_error`, so those need to be added to this enum.